### PR TITLE
[IMP] website_event: check registrable events

### PR DESCRIPTION
**Pull request by kea14**
base_automation with a trigger on_time on the start date field

### CODE REVIEW

File: addons/website_event/models/event.py [hunk 1]
```diff
@@ -156,3 +156,10 @@ def _default_website_meta(self):
         res['default_twitter']['twitter:card'] = 'summary'
         res['default_meta_description'] = self.date_begin
         return res
+
+    @api.model
+    def update_events_state(self):
+        current_company = self.env.company_id
+        now = fields.Datetime.now()
+        for event in self.search([('company_id', '=', current_company.id), ('state', '!=', 'done'), ('date_begin', '<=', now)]):
+            event.state = 'done'
```
Comment (qsm-odoo) [inline addons/website_event/models/event.py]: You are writing on each event individually, why not on all events you found directly ?
```python
events = self.search(...)
events.write({'state': 'done'})
```

Also, is there a reason to do this only for the current company ? I suppose JKE/RDE told you to do it ?

Comment (kea14) [reply addons/website_event/models/event.py]: > You are writing on each event individually, why not on all events you found directly ?

Yes, indeed.

> Also, is there a reason to do this only for the current company ? I suppose JKE/RDE told you to do it ?

Well, in fact as I saw the explanation from YTI about the new company system, I copied what he did to be able to work with the right company we are on. No ?

Comment (qsm-odoo) [reply addons/website_event/models/event.py]: Mmh. Don't know, @JKE-be will check anyway. Adapt the events.write stuff, we will see afterwards.

Comment (kea14) [reply addons/website_event/models/event.py]: > You are writing on each event individually, why not on all events you found directly ?
> events = self.search(...)
> events.write({'state': 'done'})

done

File: addons/website_event/models/event.py [hunk 2]
```diff
@@ -156,3 +156,9 @@ def _default_website_meta(self):
         res['default_twitter']['twitter:card'] = 'summary'
         res['default_meta_description'] = self.date_begin
         return res
+
+    @api.model
+    def _update_events_state(self):
+        now = fields.Datetime.now()
+        events = self.search([('state', '!=', 'done'), ('date_begin', '<=', now)])
+        events.sudo().write({'state': 'done'})
```
Comment (JKE-be) [inline addons/website_event/models/event.py]: not sudo here

And cron will call it with privilege user btw

File: addons/event/tests/test_event_flow.py [hunk 3]
```diff
@@ -135,4 +135,9 @@ def test_event_date_range(self):
         self.event_0.date_begin = datetime.datetime(2016, 1, 25, 6, 0)
         self.assertEqual(self.event_0.registration_ids.get_date_range_str(), u'next month')
 
+        self.mock_datetime.now.return_value = datetime.datetime(2019, 6, 5, 12, 0)
+        self.event_0.date_begin = datetime.datetime(2019, 6, 4, 12, 0)
+        self.event_0.date_end = datetime.datetime(2019, 6, 6, 12, 0)
+        self.assertEqual(self.event_0.registration_ids.get_date_range_str(), u'today')
```
Comment (JKE-be) [inline addons/event/tests/test_event_flow.py]: what are you testing ??????????

Comment (JKE-be) [reply addons/event/tests/test_event_flow.py]: your new method is _update_events_state ...

**Comment by JKE-be** [comment]
@sbuhl 

Are we really sure that we want this feature... ?

Maybe that end user want to do some other thing before to consider an event closed (e.g.: campaign mailing, ...)

And it is only relevant for db with many event, and this code could be done as a custo quickly

**Comment by kea14** [comment]
(reminder: master red atm = branch red)

**Comment by sbuhl** [comment]
@JKE-be Sorry for the delay, 
Does the status "Done" avoid to create a campaign mailing? 
If the event is passed and stays open, customer are still able to register for the event.
The task is mainly to avoid this behavior.

**Comment by JKE-be** [comment]
no

But mark as done means that you finish all things to do with this event imo.

You can also just avoid that user register to an event if it is in the past...
There are already a function to do it: 
`def _is_event_registrable` by commit 78014ff

Just need to do the correct code.

**Comment by kea14** [comment]
@JKE-be @sbuhl 
So, I guess this is what we want now ?

File: addons/event/models/event.py [hunk 4]
```diff
@@ -340,7 +340,7 @@ def mail_attendees(self, template_id, force_send=False, filter_func=lambda self:
 
     @api.multi
     def _is_event_registrable(self):
```
Comment (JKE-be) [inline addons/event/models/event.py]: need to call super in website_event

Comment (kea14) [reply addons/event/models/event.py]: is it like that that I should do it ?

**Comment by JKE-be** [comment]
@kea14 

why ?   :'(

website_event call _is_event_registrable()
So, event_sale that override _is_event_registrable need to call super to be sure that your original condition is true ... and not only that there are no ticket for example...

File: addons/event/tests/test_event_flow.py [hunk 5]
```diff
@@ -136,3 +136,18 @@ def test_event_date_range(self):
         self.assertEqual(self.event_0.registration_ids.get_date_range_str(), u'next month')
 
         self.patcher.stop()
+
+    def test_event_is_registrable(self):
```
Comment (JKE-be) [inline addons/event/tests/test_event_flow.py]: please, write your test un even_Sale, and add use case whith ticket, and product_id archived.

If your test was green before, when you call super and do nothing with it, it is the proof that you test don't cover all the cases

**Comment by JKE-be** [comment]
See https://github.com/odoo/odoo/compare/7071e8eaf011e894214fdb4a0524d7b1ce4eae04..6c53eef713398c9005113217d995da07fd3b8dc1

v13 is only for python3 no need to add retrocompatibility
all([]) == True 

@robodoo r+

**Comment by JKE-be** [comment]
@robodoo r+

**Comment by JKE-be** [comment]
@robodoo retry

**Pull request closed without merging**